rpmlint/checks/FilelistCheck.py

```diff
@@ -0,0 +1,144 @@
+                        r = fnmatch.translate(pattern)
+                        check['Good'][i] = re.compile(r)
+
+            if 'Bad' in check:
+                for i in range(len(check['Bad'])):
+                    pattern = check['Bad'][i]
+                    if '*' in pattern:
+                        r = fnmatch.translate(pattern)
+                        check['Bad'][i] = re.compile(r)
+
+    def check(self, pkg):
+        if pkg.isSource():
+            return
+
+        files = pkg.files()
+        for check in self.checks:
+            if 'IgnorePkgIf' in check:
+                if getattr(self, check['IgnorePkgIf'])(pkg):
+                    continue
+
+            if 'Good' in check or 'Bad' in check:
+                for f in files:
+                    ok = False
+                    if 'Good' in check:
+                        for g in check['Good']:
+                            if ((not isinstance(g, str) and g.match(f)) or
+                                    g == f):
+                                ok = True
+                                break
+                    if ok:
+                        continue
+
+                    if 'Bad' in check:
+                        for b in check['Bad']:
+                            if 'IgnoreFileIf' in check:
+                                if getattr(self, check['IgnoreFileIf'])(pkg, f):
+                                    continue
+                            if ((not isinstance(b, str) and b.match(f)) or
+                                    b == f):
+                                self.output.add_info('E', pkg, check['Message'], f)
+
+        invalidfhs = set()
+        invalidopt = set()
+
+        is_suse = (pkg.header[RPMTAG_VENDOR] and 'SUSE' in byte_to_string(pkg.header[RPMTAG_VENDOR]))
+
+        # the checks here only warn about a directory once rather
+        # than reporting potentially hundreds of files individually
+        for f, pkgfile in files.items():
+            type = (pkgfile.mode >> 12) & 0o17
+
+            # append / to directories
+            if type == 4:
+                f += '/'
+
+            if not f.startswith(self.goodprefixes):
+                base = f.rpartition('/')
+                pfx = None
+                # find the first invalid path component
+                # (/usr/foo/bar/baz -> /usr)
+                while (base[0] and not base[0].startswith(self.goodprefixes) and
+                       not base[0] in self._restricteddirs):
+                    pfx = base[0]
+                    base = base[0].rpartition('/')
+
+                if not pfx:
+                    invalidfhs.add(f)
+                else:
+                    invalidfhs.add(pfx)
+
+            if f.startswith('/opt'):
+                try:
+                    provider = f.split('/')[2]
+                except Exception:
+                    continue
+                if is_suse and (provider == 'suse' or provider == 'novell'):
```

Comment:
Any idea what for is this is_suse thingy?

Reply:
Dunno. I would take is as-is.

Reply:
It's an exception for `/opt`. If the package has `SUSE` or `Novell` and the host system running rpmlint is a SUSE system, `/opt` rules don't apply.

Reply:
ok, lets keep it ugly as-is.